### DISCUSSION TITLE
Help install component relation picker with filter

### DISCUSSION THREAD

**Discussion by harmkroog** [Budibase/budibase#10505] [Help] [answered]
Hi,
iam running budibase docker image.
ive found this thread here. https://github.com/Budibase/budibase/pull/8897
But i dont understand how i can install this.
Can anyone help me, because relationship picker with filter would meeeegaaaaa ;-)

Thank you!
Best regards Bonkersdeluxe

**Comment by ConorWebb96** [comment, marked as answer]
Hey @harmkroog,

I don't think this was ever merged in, the enhancement request can be found here: https://github.com/Budibase/budibase/issues/7810. Feel free to comment under it and that may bump it in priority.

This functionality exists in the relationship picker within the data view. We hope to eventually include the same functionality to the relationship picker within design view as well.

I hope this makes sense.

**Comment by harmkroog** [comment]
Hi ConorWebb96,
Thank you!
I dont understand what you mean with dataview?

my problem is following:

Ive got an main Table with customers
![Auswahl_121](https://github.com/Budibase/budibase/assets/130768693/be217d36-2bbc-4ce3-8181-5f10e9f426e6)


If i klick on Adresses i can list and create adresses for this customer.
This works fine.
List of addresse from selected customer
![Auswahl_120](https://github.com/Budibase/budibase/assets/130768693/630055ee-ca90-4767-9054-d180394104fd)


If i klick on Verträge, an new table with all abos will list.
(The most field are empty because i design the gui and the functionalaty)
This customer get only one  abo
![Auswahl_118](https://github.com/Budibase/budibase/assets/130768693/acaf30ae-6d2a-4ac3-8707-54ab4963b82a)

I klick on the abo and now my problem comes out
If i select from the relationpciker the Adresses all Adresses will be show and not only the addresses from the customer.
![Auswahl_119](https://github.com/Budibase/budibase/assets/130768693/1987fdc6-bf7e-4398-b6dd-05a2aaf56ec7)

How can i resolve this? I save the the customer ID in an state variable.
In the dataprovider i can filter it throw the state variable
But in Dataview i cant filter states or can i do so?
Thank you very much!
Best regards Bonkersdeluxe

**Reply by ConorWebb96** [reply]
Hey @harmkroog,

When I mentioned data view and design view I'm talking about these tabs within the screenshot below.
<img width="995" alt="Screenshot 2023-05-11 at 07 34 45" src="https://github.com/Budibase/budibase/assets/126772285/c99ff93c-d21b-4d17-a96d-f9d63d0656d6">

State is only available to use within the design view you can't use it within the data view. The only method of filtering i'm aware of is the dynamic filter which can be used via clicking the filter icon above the tables: screenshot below.
<img width="968" alt="Screenshot 2023-05-11 at 07 36 17" src="https://github.com/Budibase/budibase/assets/126772285/fce504c9-9613-43b6-9b2f-c879f2fac7bf">

Hopefully this makes sense!

**Comment by harmkroog** [comment]
Hi,
Thank you for answer!
but i dont understand how i filter it  in design view.
Sorry i describe it wrong.

What would i reach.

In an screen i have an table with contracts.
If i click on contract i can change the Adressfield. 
But this Adressfield lists all adresses from all customer.
I wish, i could filter it , to show only adresses from the customer that owns the contract.
![237111271-1987fdc6-bf7e-4398-b6dd-05a2aaf56ec7](https://github.com/Budibase/budibase/assets/130768693/1608c542-bc76-41cb-91ef-b6e9eb288c62)


If i could to this with the filter in data view, i dont know what variables i should use, because you say state variables are only available in design view.

Thank you!
Best regards Bonkersdeluxe

**Comment by harmkroog** [comment]
Hi, have nobody no idea how to solve this?
Iam very happy if anybody has an solution for me.
Thank you!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment]
Ive solved this with this article. https://github.com/Budibase/budibase/discussions/4579
i dont understand this with Dataview....
This an workaround but it works for me
I hope they will merged Relationship Picker with filter in the future.
That makes that really easier, define your filter. Done.

Thank you!
Best regards Bonkersdeluxe
